packages/patterns/src/types.js

```diff
@@ -574,37 +573,25 @@ export {};
  */
 
 /**
- * @typedef {{
- *   klass: 'methodGuard',
- *   callKind: 'sync' | 'async',
- *   argGuards: ArgGuard[]
- *   optionalArgGuards?: ArgGuard[]
- *   restArgGuard?: Pattern
- *   returnGuard: Pattern
+ * @typedef {CopyTagged & {
+ *   [Symbol.toStringTag]: 'guard:methodGuard',
+ *   payload: {
+ *     callKind: 'sync' | 'async',
+ *     argGuards: ArgGuard[]
+ *     optionalArgGuards?: ArgGuard[]
+ *     restArgGuard?: Pattern
+ *     returnGuard?: Pattern
+ *   }
  * }} MethodGuard
- *
- * TODO https://github.com/endojs/endo/pull/1712 to make it into a genuine
- * guard that is distinct from a copyRecord.
- * Once we're ready for such a compat break, we might also take the
- * opportunity to rename `restArgGuard` and `returnGuard`
- * to reflect that their value must be a Pattern rather that a
- * non-pattern guard.
```

Comment:
Any desire to take on this change as well?

Reply:
I'd like to. Depends on how we cope with the compat issue shown at https://github.com/endojs/endo/pull/1715#issuecomment-1682929677

Reply:
If all we do to smooth the transition is https://github.com/endojs/endo/pull/1745 , then we would not enable this renaming. I think that's ok.

Reply:
All we did is #1745

Reply:
Resolving, with these renamings not done.